### ISSUE TITLE
Extra metadata field created after editing a Dataset

### ISSUE THREAD

**Issue by raprasad** [IQSS/dataverse#3193]
### Description

Within the citation metadata block, the following fields were created in the _database_ **without the user entering information**:
- Related Publication (DatasetField)
  - ID Type (child DatasetField under "Related Publication")
    - No value was assigned to "ID Type"
- Contributor (Dataset Field)
  - Contributor Type (child DatasetField under "Contributor")
    - No value was assigned to "Contributor Type"
### Important Note

**Note**: These fields were created in the database but do not appear in the UI--assumedly because no value was assigned to "ID Type" or "Contributor Type"
### Reproducing the bug

Sequence that created the extra DatasetFields:
1.  Create Dataset with minimal metadata
2.  Save Draft of Dataset
3.  Edit Dataset
4.  Change Title of Dataset
5.  Save Draft of Dataset

Note: The Dataset does not need to be published.
### Looking for the cause

The bug may be due to both these of the parent fields ("Related Publication" and "Contributor") having their first sub field being a drop down with a controlled vocabulary.

This pattern with controlled vocabulary fields also holds true for other metadata blocks.  For example:
- **Geospatial metadata block**.  If any data is filled in, these two DatasetField are created, even if blank:
  - Geographic Coverage -> Country / Nation subfield
- **Journal metadata block**.  If any data is filled in, the "Type of Article" DatasetField will be created.
  - This is also a controlled vocabulary field.

**Comment by pdurbin** [comment]
This smells a bit like the "smurfFunder" bug I reported at #973.

**Comment by djbrooke** [comment]
@jggautier - low priority, can you see if this still occurs?

**Comment by jggautier** [comment]
Yes, I can see in the Demo Dataverse database and Harvard Dataverse database that this still occurs. Not sure how much of a problem it is.

If I create a dataset, and the dataset create form shows metadata fields that use drop down menus, and I don't select values for those fields, the database still records a datasetfield id for that field. The most common cases are the author identifier scheme field and related publication ID type field, which are both shown on the dataset create page:

<img width="719" alt="Screen Shot 2019-10-29 at 10 05 19 PM" src="https://user-images.githubusercontent.com/18374574/67823182-723bb780-fa98-11e9-861a-acbbb99ddf1d.png">

<img width="720" alt="Screen Shot 2019-10-29 at 10 05 29 PM" src="https://user-images.githubusercontent.com/18374574/67823160-5c2df700-fa98-11e9-931e-0f93c31ca559.png">

The following screenshot shows IDs recorded for datasetfields in Demo Dataverse after I created a dataset and left the four related publication fields empty:

<img width="879" alt="after-creating-dataset" src="https://user-images.githubusercontent.com/18374574/67823391-4967f200-fa99-11e9-9919-dcca6ab105fe.png">

Dataverse still records an ID for the "publication" compound field.

When I go to edit the metadata of my new draft dataset, the additional fields shown can include other fields with drop down menus, like the contributor field in the "Citation" metadata block that ships with Dataverse. When I leave those fields blank, and save the dataset draft again, the database records a datasetfield ID for those newly shown dataset fields, too:

<img width="880" alt="after-editing-dataset-metadata" src="https://user-images.githubusercontent.com/18374574/67823810-de1f1f80-fa9a-11e9-89f6-5823eb8603bb.png">

All 21 additional rows in this screenshot are database IDs created for fields that contain dropdown menus, all of which I left empty. Most are from a few of Harvard Dataverse's custom metadata blocks.

Are these IDs just taking up space or are they needed for some other reason? Most (or all?) of the 21 IDs in that screenshot are referring to the parent field (like the related publication compound field) and not the specific child field (like the related publication ID type within the related publication compound field), so I'm not sure if IDs for the child fields are also being created in other tables in the database. The values for the controlled vocabularies in the drop down menus are stored in a controlledvocabularyvalue table, but there are no null or empty values in that table in Harvard Dataverse's database. Not sure where else to look or how.

**Comment by jggautier** [comment]
I've been reviewing GitHub issues assigned to me and re-tested this one on Demo Dataverse, which is now using version 5.12. What I described in my last comment is still occurring.

After reading Ramen's original comment again, I'm not sure that what I wrote completely covers what Ramen described.

When Danny asked me to see if it was still happening, he wrote that this was low priority. Why is this low priority? Is it simply creating unused rows in database tables?

**Comment by cmbz** [comment]
To focus on the most important features and bugs,     we are closing issues created before 2020 (version 5.0)     that are not new feature requests with the label 'Type: Feature'.    

If you created this issue and you feel the team     should revisit this decision, please reopen the issue and leave a comment.

**Comment by sbarbosadataverse** [comment]
We will have the community assist in testing to see if this is still happening. If no feedback within two weeks, I will close the issue until a user reports it again.

**Comment by cmbz** [comment]
2025-07-28
- Closing as per comment: https://github.com/IQSS/dataverse/issues/3193#issuecomment-2719172784
- Please open a new issue if you are still experiencing this problem.
